src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -791,14 +792,18 @@ private async Task SendPingAckAsync(ReadOnlyMemory<byte> pingContent)
         {
             Debug.Assert(pingContent.Length == FrameHeader.PingLength);
 
+            // Copy pingContent before we go async so the caller can
+            // discard their buffer without waiting for us to complete.
+            long pingContentLong = BitConverter.ToInt64(pingContent.Span);
```

Comment:
Is there a particular reason to use BitConverter here, as opposed to BinaryPrimitives.ReadUInt64BigEndian? We use BinaryPrimitives elsewhere in the code. Seems like we should be consistent.

Reply:
I actually haven't found any instances of `BinaryPrimitives` being used to shuttle opaque bytes, so this isn't a consistency issue. It made sense to avoid byte shuffling when endianness concerns could be kept to a single small method.